.ci/jobs/release-docker.yml

```diff
@@ -0,0 +1,34 @@
+---
+
+- job:
+    name: "elastic+rally-release-docker+master"
+    display-name: "elastic / rally # master - release docker"
+    description: "Release Docker Artifacts for Rally"
+    parameters:
+      - string:
+          name: branch_specifier
+          default: "refs/heads/master"
+          description: "the Git branch specifier to build (&lt;branchName&gt;, &lt;tagName&gt;, &lt;commitId&gt;, etc.)"
+      - string:
+          name: RELEASE_VERSION
+          default: ""
+          description: "The version to release e.g. '1.2.1'."
+    triggers: []
+    builders:
+      - shell: |
+          #!/usr/bin/env bash
+          set -eo pipefail
+          set +x
+          VAULT_TOKEN=$(vault write -field=token auth/approle/login role_id="$VAULT_ROLE_ID" secret_id="$VAULT_SECRET_ID")
```

Comment:
I think we need to test / look a bit deeper here.

For ref check these internal links:

[link 1](https://github.com/elastic/infra/issues/12089#issuecomment-499159694)
[link 2](https://github.com/elastic/infra/blob/9766bf6c3a8d2dde614aff1cfd6e51c21b50677f/ci/jjb/elasticsearch-ci/globals/macros/wrappers.yml#L8-L13)
[link 3](https://github.com/elastic/infra/issues/32842)

and

[link 4](https://github.com/elastic/infra/tree/master/flavortown/jjbb#how-does-it-work)